cartridges/bm_affirm/cartridge/scripts/handleTransaction.js

```diff
@@ -0,0 +1,274 @@
+"use strict"
+
+/**
+ * Affirm Transaction Handler
+ */
+
+/* API Includes */
+var OrderMgr = require("dw/order/OrderMgr")
+var Transaction = require("dw/system/Transaction")
+var Resource = require("dw/web/Resource")
+
+/* Script Modules */
+var Utils = require("*/cartridge/scripts/util/affirmUtils")
+
+/* Custom Logger */
+var LogUtils = require("*/cartridge/scripts/util/logger")
+var log = LogUtils.getLogger("Affirm-HandleTransaction")
+
+/* Affirm API */
+var api = require("*/cartridge/scripts/api/affirmAPI")
+
+/**
+ * Read Transactions history from Affirm API and update the order custom attribute
+ * @param {string} transactionID - Affrim Transaction ID
+ */
+function read(transactionID) {
+	var Order = OrderMgr.searchOrder(
+		"custom.AffirmExternalId = {0}",
+		transactionID
+	)
+
+	try {
+		var _r = api.read(transactionID)
+		var events = []
+		if (typeof _r.response !== 'undefined' && typeof _r.response.events !== 'undefined') {
+			// Truncate if the count of records is larger than 25
+			events = _r.response.events.length > 25
```

Comment:
why do we have this limit hard coded?

Reply:
We're running in to the character limit for custom attribute here.